packages/react-topology/src/elements/BaseGraph.ts

```diff
@@ -73,6 +85,27 @@ export default class BaseGraph<E extends GraphModel = GraphModel, D = any> exten
     this.scaleExtent = scaleExtent;
   }
 
+  setDetailsLevelSettings(settings: ScaleDetailsSettings): void {
+    this.scaleDetailsSettings = settings;
+    let detailsLevel = ScaleDetailsLevel.high;
+    if (this.scaleDetailsSettings) {
+      if (this.scale <= this.scaleDetailsSettings[ScaleDetailsLevel.low]) {
+        detailsLevel = ScaleDetailsLevel.low;
+      } else if (this.scale <= this.scaleDetailsSettings[ScaleDetailsLevel.medium]) {
+        detailsLevel = ScaleDetailsLevel.medium;
+      }
+    }
+    this.setDetailsLevel(detailsLevel);
+  }
+
+  getDetailsLevel(): ScaleDetailsLevel {
+    return this.detailsLevel;
```

Comment:
This should be a computed value based on scale.
If we want to support overriding the value it should then fallback to a private observable that always takes precedence when set.

Reply:
Lets add the override if we determine it is wanted and remove the `setDetailsLevel` for now.